capi_problems.rst

```diff
@@ -94,12 +99,99 @@ all types of API users:
 * Manage sub-interpreters
 * Handle and send signals
 
+External users often support multiple Python versions
+or Python implementations. For them, these are the common
+requirements:
+
+* Specification/documentation of the API and ABI
+* Backwards/forwards compatibility and stability for:
+    * API (code works as-is but requires recompilation)
+    * ABI (compiled binaries work)
+    * Across CPython versions
+    * Across Python implementations
+
+Note: in practice, there will be tradeoffs between some of the requirements,
+for example, stability vs. performance.
+
 Groups of users are:
 
+**CPython developers**
+
+CPython developers change the CPython internals to implement new
+functionality or improve/optimize the current functionality. While ideally,
+they would be free to alter any internal details, in practice, anything that
+is intentionally (or even unintentionally) exposed to external C API users
+should remain compatible (depending on the concrete policy for the
+specific API). If it is part of the Stable ABI, it must remain binary
+compatible.
+
+One recent notable example is that a significant amount of work on the
+Gilectomy project and the `PEP 703 <https://peps.python.org/pep-0703/>`__
+was spent on dealing with reference counting, most specifically on
+keeping the reference counting contract, which is only exposed in C API,
+while making it scale with parallel execution.
+
+For more details see the Discuss
+`thread <https://discuss.python.org/t/lets-get-rid-of-the-stable-abi-but-keep-the-limited-api/18458>`__:
+Let’s get rid of the stable ABI, but keep the limited API.
+
+Because the CPython code is by definition intended to run only on given
+version of CPython and is always recompiled, the following requirements
+*do not apply* to CPython developers:
```

Comment:
Can we reword this somehow? Everyone's requirement apply to cpython core devs because they are the stewards of the language.

Reply:
yes, absolutely valid point. What about "...do not apply to the C API usage in CPython core codebase"?